docs: document how external link icons work

ajitsinghkaler opened the pull request:
After PR #36601 which added icons to all external links. Documented how this is happening via comments in scss file. For details visit PR #36601

@IgorMinar suggested that we should add documentation on how external link icons work. For details see https://github.com/angular/angular/pull/36601#issuecomment-625952724

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
No documentation on how external link icons work

Issue Number: N/A


## What is the new behavior?
No documentation on how external link icons work

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

ajitsinghkaler commented:
@IgorMinar can you please look into this

ajitsinghkaler commented:
@IgorMinar can you please look into this

ajitsinghkaler commented:
@IgorMinar can yo please look into this

IgorMinar commented on aio/src/styles/0-base/_typography.scss:
```diff
@@ -193,6 +193,12 @@ code {
 
+// The following css adds an icon to external icons in the docs area. The following classes
+// .folder-api, .folder-cli, .folder-docs,.folder-guide, .folder-start, .folder-tutorial
+// contains all the documentation in the aio website.
```
```suggestion
```

IgorMinar commented on aio/src/styles/0-base/_typography.scss:
```diff
@@ -193,6 +193,12 @@ code {
 }
 
+// The following css adds an icon to external icons in the docs area. The following classes
+// .folder-api, .folder-cli, .folder-docs,.folder-guide, .folder-start, .folder-tutorial
```
```suggestion
```

IgorMinar commented on aio/src/styles/0-base/_typography.scss:
```diff
@@ -193,6 +193,12 @@ code {
+// The following css adds an icon to external icons in the docs area. The following classes
+// .folder-api, .folder-cli, .folder-docs,.folder-guide, .folder-start, .folder-tutorial
+// contains all the documentation in the aio website.
+// Added an icon in `:after` of all the links which contain http or https. These define all external links.
```
```suggestion
// We add the icon to all external links which are identified as absolute links (those that start with `http` or https`).
```

IgorMinar commented on aio/src/styles/0-base/_typography.scss:
```diff
@@ -193,6 +193,12 @@ code {
+// .folder-api, .folder-cli, .folder-docs,.folder-guide, .folder-start, .folder-tutorial
+// contains all the documentation in the aio website.
+// Added an icon in `:after` of all the links which contain http or https. These define all external links.
+// For Details visit PR #36601
```
```suggestion
// For more info see PR #36601
```

IgorMinar commented on aio/src/styles/0-base/_typography.scss:
```diff
@@ -213,6 +219,9 @@ code {
       }
     }
 
+    // Docs body contain links to github which are icons these are also external links
```
```suggestion
    // The docs-viewer also contain links to GitHub (e.g. the edit this page icon) identified with `.github-links` class.
```

IgorMinar commented on aio/src/styles/0-base/_typography.scss:
```diff
@@ -213,6 +219,9 @@ code {
     }
 
+    // Docs body contain links to github which are icons these are also external links
+    // Don't want to add icon to these links. So icons are hidden on these links   
```
```suggestion
    // We don't want to add the external link icon to these links, so we hide them.
```

IgorMinar commented on aio/src/styles/0-base/_typography.scss:
```diff
@@ -213,6 +219,9 @@ code {
 
+    // Docs body contain links to github which are icons these are also external links
+    // Don't want to add icon to these links. So icons are hidden on these links   
+    // For Details visit PR #36601
```
```suggestion
```

ajitsinghkaler committed:
docs: document how external link icons work

After PR #36601 which added icons to all external links. Documented how this is happening via comments in scss file. For details visit PR #36601

IgorMinar reviewed (approved):
LGTM now. thanks!
